apps/test/unit/templates/rubrics/LearningGoalsTest.jsx

```diff
@@ -39,9 +55,173 @@ describe('LearningGoals', () => {
+    });
+
+    it('returns a result that does not contain the comments', () => {
+      const result = getAnonymizedCode();
+      expect(result).to.not.have.string('// add them together');
+      expect(result).to.not.have.string('var z = x + y');
+      expect(result).to.not.have.string('// code');
+    });
+
+    it('returns a result that does still contain the code', () => {
+      const result = getAnonymizedCode();
+      expect(result).to.have.string('var x = 5');
+      expect(result).to.have.string('var y = 6');
+    });
+  });
+
+  describe('annotateLines', () => {
+    it('should do nothing if the AI observation does not reference any lines', () => {
+      // The AI tends to misreport the line number, so we shouldn't rely on it
+      annotateLines('This is just a basic observation.');
+      expect(annotateLineStub.notCalled).to.be.true;
+    });
+
+    it('should annotate a single line of code referenced by the AI', () => {
+      // The AI tends to misreport the line number, so we shouldn't rely on it
+      annotateLines('Line 1: This is a line of code `var x = 5;`');
+      sinon.assert.calledWith(annotateLineStub, 'This is a line of code', 2);
+    });
+
+    it('should annotate the first line of code referenced by the AI', () => {
+      annotateLines('Line 1: This is a line of code `var x = 5; var y = 6;`');
+      sinon.assert.calledWith(annotateLineStub, 'This is a line of code', 2);
+    });
+
+    it('should highlight a single line of code referenced by the AI', () => {
+      // The AI tends to misreport the line number, so we shouldn't rely on it
+      annotateLines('Line 1: This is a line of code `var x = 5; var y = 6;`');
+      sinon.assert.calledWith(highlightLineStub, 2);
+    });
+
+    it('should highlight all lines of code referenced by the AI', () => {
+      annotateLines('Line 1: This is a line of code `var x = 5; var y = 6;`');
+      sinon.assert.calledWith(highlightLineStub, 2);
+      sinon.assert.calledWith(highlightLineStub, 3);
+    });
+
+    it('should just highlight the lines the AI thinks if the referenced code does not exist', () => {
+      annotateLines('Line 45: This is a line of code `var z = 0`');
+      sinon.assert.calledWith(annotateLineStub, 'This is a line of code', 45);
+      sinon.assert.calledWith(highlightLineStub, 45);
+    });
+
+    it('should annotate the last line of code when referenced by the AI', () => {
+      annotateLines('Line 55: This is a line of code `draw();`');
+      sinon.assert.calledWith(annotateLineStub, 'This is a line of code', 8);
+    });
+
+    it('should highlight the last line of code when referenced by the AI', () => {
+      annotateLines('Line 55: This is a line of code `draw();`');
+      sinon.assert.calledWith(highlightLineStub, 8);
+    });
+
+    it('should ignore code snippets that are empty', () => {
+      annotateLines('Line 42: This is totally a thing ` `');
+      sinon.assert.notCalled(highlightLineStub);
+    });
+  });
```

Comment:
can you please add a test covering the `Line x-y: ...` case?

Reply:
Added a test covering that case. Good catch.